### ISSUE TITLE
Install/Uninstall process

### ISSUE THREAD

**Issue by johnsimons** [Particular/NServiceBus#1522]
@BogdanMitrache in the NServiceBus installer we need the following logic for installing/uninstalling:
- User install v1.0.0 (first new install)
- User downloads a new patch release v1.0.1
- When user installs v1.0.1, version 1.0.0 should be uninstalled automatically and v1.0.1 installed
- If user later on downloads v1.1 or v2.0
- When user installs v1.1 or v1.2 then the installer will install both of those side by side (so v1.0.1 would **not be uninstalled!**)

So in summary, given a version like x.y.z:
- Any changes in Z triggers an uninstall of any previous x.y.# automatically
- Any changes to either x or y will trigger a side by side install

I hope this makes sense?
